src/Riok.Mapperly/Descriptors/MappingBodyBuilders/NewInstanceObjectMemberMappingBodyBuilder.cs

```diff
@@ -66,6 +66,19 @@ private static void BuildInitOnlyMemberMappings(INewInstanceBuilderContext<IMapp
 
             BuildInitMemberMapping(ctx, targetMember, sourceMemberPath);
         }
+
+        var initOnlyIgnoredTargetMembers = includeAllMembers
+            ? ctx.IgnoredTargetMembers.Values.ToArray()
+            : ctx.IgnoredTargetMembers.Values.Where(x => x.CanOnlySetViaInitializer()).ToArray();
+        foreach (var targetMember in initOnlyIgnoredTargetMembers)
+        {
+            ctx.IgnoredTargetMembers.Remove(targetMember.Name);
+
+            if (ctx.MemberConfigsByRootTargetName.Remove(targetMember.Name, out var memberConfigs))
+            {
+                BuildInitMemberMapping(ctx, targetMember, memberConfigs);
+            }
+        }
```

Comment:
Could you add a comment what this does and why it is needed? Does this work nicely with required init properties?

Reply:
Added comment and added required/init tests